modules/core/src/matrix.cpp

```diff
@@ -2662,6 +2666,10 @@ void _OutputArray::assign(const Mat& m) const
     {
         *(Mat*)obj = m;
     }
+    else if (k == MATX)
```

Comment:
MATX assumes constant size and constant type. Not sure that getMat() will do the right thing if the assigned matrix has a different size and/or different type. Use m.copyTo(getMat()) instead.